projects/out_of_sample/scripts/create_report.py

```diff
@@ -0,0 +1,315 @@
+        mapper = fsspec.get_mapper(diags_url)
+        diags.to_zarr(mapper, mode="w", consolidated=True)
+        print(f"Saved online novelty data to {diags_url}.")
+    return diags
+
+
+def make_diagnostic_plots(
+    report_sections, temp_output_dir, models, model_diags, grid, type
+):
+    """
+    Fills in sections of a future html report for a collection of models with
+    diagnostic data. The first section is a time plot that compares the fraction
+    of novelties over time for each model on the target dataset. The remaining
+    sections are for each model, which includes 4 plots: (1) a map plot of the
+    average rate of novelties over time, (2) a Hovmoller plot of the novelty fraction
+    by time and latitude, (3) the fraction of novelties for each as a function
+    of the chosen cutoff, and (4) a histogram of novelty scores.
+    """
+    type_file_name = type.lower().replace(" ", "-")
+    fig = plt.figure()
+    for model in models:
+        diags = model_diags[model.name]
+        diags.is_novelty.mean(["tile", "x", "y"]).plot(label=model.name)
+    plt.legend()
+    plt.title("Fraction of novelties over time")
+    report.insert_report_figure(
+        report_sections,
+        fig,
+        filename=f"novelties-over-time-{type_file_name}.png",
+        section_name=f"{type} novelty frequency over time",
+        output_dir=temp_output_dir.name,
+    )
+    plt.close(fig)
+
+    for model in models:
+        section_name = f"{type} novelty frequency of {model.name}"
+        diags = model_diags[model.name]
+
+        print(model.name)
+
+        fig, _, _, _, _ = fv3viz.plot_cube(
+            ds=diags.is_novelty.mean("time")
+            .to_dataset(name="novelty_frac")
+            .merge(grid),
+            var_name="novelty_frac",
+            vmax=1,
+            vmin=0,
+        )
+        plt.title("Time-averaged fraction of novelties")
+        report.insert_report_figure(
+            report_sections,
+            fig,
+            filename=f"time-avg-{model.name}-{type_file_name}.png",
+            section_name=section_name,
+            output_dir=temp_output_dir.name,
+        )
+        plt.close(fig)
+
+        fig = plt.figure()
+        vcm.zonal_average_approximate(grid.lat, diags.is_novelty).plot(x="time")
+        plt.title("Hovmoller plot of fraction of novelties")
+        report.insert_report_figure(
+            report_sections,
+            fig,
+            filename=f"hovmoller-{model.name}-{type_file_name}.png",
+            section_name=section_name,
+            output_dir=temp_output_dir.name,
+        )
+        plt.close(fig)
+
+        fig = plt.figure()
+        if isinstance(model.nd, OCSVMNoveltyDetector):
+            min_score = diags.novelty_score.min().values
+            cutoffs = np.linspace(min_score, 0, 20)
+        elif isinstance(model.nd, MinMaxNoveltyDetector):
+            max_score = diags.novelty_score.max().values
+            cutoffs = np.linspace(0, max_score, 20)
```

Comment:
Maybe the most natural thing is to have a range of from `min_score` to `max_score` for any novelty detector? It seems like that should work for either. (Also, we won't have sign issues once we start using `centered_score`, but none of our generated datasets have that field yet, since that change just got pushed.)